modules/network/src/integrationTest/java/org/apache/ignite/network/scalecube/ItScaleCubeNetworkMessagingTest.java

```diff
@@ -320,6 +323,69 @@ public void testMessageGroupsHandlers(TestInfo testInfo) throws Exception {
         assertThat(networkMessageFuture, willBe(equalTo(networkMessage)));
     }
 
+    /**
+     * Makes sure that a node that dropped out from the Physical Topology cannot reappear with same ID.
+     *
+     * @throws Exception in case of errors.
+     */
+    @Test
+    public void nodeCannotReuseOldId(TestInfo testInfo) throws Exception {
+        testCluster = new Cluster(3, testInfo);
+
+        testCluster.startAwait();
+
+        String outcastName = testCluster.members.get(testCluster.members.size() - 1).nodeName();
+
+        knockOutNode(outcastName);
+
+        CountDownLatch appeared = reanimateNode(outcastName);
+
+        assertFalse(appeared.await(3, TimeUnit.SECONDS), "Node reappeared");
```

Comment:
So this test is always 3 seconds, right? Can we maybe listen for some event like "Node failed to join because of the stale id"? I see we have a FailureHandler, that can be used for that, right?